aws-xray-recorder-sdk-core/src/main/java/com/amazonaws/xray/emitters/UDPEmitter.java

```diff
@@ -58,17 +58,17 @@ public boolean sendSubsegment(Subsegment subsegment) {
         if (logger.isDebugEnabled()) {
             logger.debug(subsegment.prettyStreamSerialize());
         }
-        return sendData((PROTOCOL_HEADER + PROTOCOL_DELIMITER + subsegment.streamSerialize()).getBytes());
+        return sendData((PROTOCOL_HEADER + PROTOCOL_DELIMITER + subsegment.streamSerialize()).getBytes(), subsegment);
     }
 
-    private boolean sendData(byte[] data) {
+    private boolean sendData(byte[] data, Entity entity) {
         DatagramPacket packet = new DatagramPacket(sendBuffer, DAEMON_BUF_RECEIVE_SIZE, config.address);
         packet.setData(data);
         try {
             logger.debug("Sending UDP packet.");
             daemonSocket.send(packet);
         } catch (IOException e) {
-            logger.error("Exception while sending segment over UDP.", e);
```

Comment:
Could you do not delete this one? From my understanding, there should have two log messages. One is log error showing the exception what happened through sending data. The second one should be log debug showing which segments/subsegments fail to be sent out.

Reply:
I kept the original message and moved the printing of the entity to the info log level.
Is that more acceptible to you?

Moving it to debug level will also hit the debug statement in `com.amazonaws.xray.emitters.UDPEmitter#sendSegment` which will print all possible traces, so then you also have no clue anymore which one was too big and failed.